src/ice-engine/threadpool.cpp

```diff
@@ -0,0 +1,58 @@
+//#include "common.h"
+#include "ice-engine/threadpool.h"
+#include <functional>
+#include <iostream>
+#include <future>
+#include <thread>
+#include <queue>
+#include <vector>
+#include <condition_variable>
+
+threadpool::threadpool(uint8_t num_threads)
+{
+    //
+    worker_threads.reserve(num_threads);
+    // initialize to be running a worker task
+    // worker task should simply use cond variable to access whether or not there is work
+    int i = 0;
+    mStop = false;
+    while(worker_threads.size() < worker_threads.capacity())
+    {
+        worker_threads.push_back(std::thread([this]
+        {
+           this->run_worker_thread();
+	}));
+	worker_threads[i].detach();
```

Comment:
There is no point in detaching threads if you are keeping track of them in a list.